Gems/ROS2/Code/Include/ROS2/Lidar/LidarRaycasterBus.h

```diff
@@ -0,0 +1,104 @@
+        //! the imperfections arising due to various physical factors e.g. fluctuations in rotary motion of the lidar (angular noise) or
+        //! distance accuracy (distance noise).
+        //! For the the details about Gaussian noise, please refer to https://en.wikipedia.org/wiki/Gaussian_noise.
+        //! You can also check-out the RobotecGPULidar (integrated in the RobotecGPULidar Gem) docs concerning these types of lidar noise at
+        //! https://github.com/RobotecAI/RobotecGPULidar/blob/v11/docs/GaussianNoise.md.
+        //! @param angularNoiseStdDev Angular noise standard deviation.
+        //! @param distanceNoiseStdDevBase Base value for Distance noise standard deviation.
+        //! @param distanceNoiseStdDevRisePerMeter Value by which the distance noise standard deviation increases per meter length from
+        //! the lidar.
+        // TODO - different starting points for rays, distance from reference point, noise models, rotating mirror sim, other
+        // TODO - customized settings. Encapsulate in lidar definition and pass in constructor, update transform.
+        virtual void ConfigureNoiseParameters(
+            float angularNoiseStdDev, float distanceNoiseStdDevBase, float distanceNoiseStdDevRisePerMeter)
+        {
+            AZ_Assert(false, "This Lidar Implementation does not support noise!");
+        }
+
+        //! Configures Layer ignoring parameters
+        //! @param ignoreLayer Should a specified collision layer be ignored?
+        //! @param layerIndex Index of collision layer to be ignored.
+        virtual void ConfigureLayerIgnoring(bool ignoreLayer, unsigned int layerIndex)
+        {
+            AZ_Assert(false, "This Lidar Implementation does not support collision layer configurations!");
+        }
+
+        //! Excludes entities with given EntityIds from raycasting.
+        //! @param excludedEntities list of entities marked for exclusion.
+        virtual void ExcludeEntities(const AZStd::vector<AZ::EntityId>& excludedEntities)
+        {
+            AZ_Assert(false, "This Lidar Implementation does not support entity exclusion!");
+        }
+
+        //! Configures max range point addition.
+        //! @param includeMaxRange Should the raycaster add points at max range for rays that exceeded their range?
+        virtual void ConfigureMaxRangePointAddition(bool addMaxRangePoints)
+        {
+            AZ_Assert(false, "This Lidar Implementation does not support Max range point addition configuration!");
+        }
+
+    protected:
+        static void ValidateRayRange(float range)
+        {
+            AZ_Assert(range > 0.0f, "Provided ray range was of incorrect value: Ray range value must be greater than zero.")
+        }
+
+        static void ValidateRayOrientations(const AZStd::vector<AZ::Vector3>& orientations)
+        {
+            AZ_Assert(!orientations.empty(), "Provided ray orientations were of incorrect value: Ray orientations must not be empty.")
+        }
```

Comment:
question: Is having these static functions on the interface a good idea? Should these just be implemented in the derived type? Not sure 100% how this will play with the EBus mechanics

Reply:
These restrictions are common to all lidar implementations which if implemented in derived types would add additional boilerplate to each Lidar implementation. If however they could cause issues with the EBus mechanics I'm open to removing them from the interface.